reactioncommerce/reaction
# Fix flash 403/404 views when auth state change

Resolve #3241 

This PR fix flashing 403/404 page  when user try to sign out or sign in.
#### Fix
This is fixed by checking if the reaction router grants permission to the user who is currently entering the page if not reveal the unauthorized. 

#### Test
- Click on product to reveal the product detail page or any other page aside the home page.
- Sign in and Sign out from the product detail page or any of the pages.
- Check to see that 403/404 page does not flash when performing sign in and out.
